scripts/checks/branch_adds_binary

```diff
@@ -0,0 +1,31 @@
+#! /bin/bash
```

Comment:
Maybe a better location for these could be `tests/travis`?

Reply:
Maybe, can I create a `tests/scripts`directory as well?
The test itself is generic, I would like to demonstrate the use of hooks and how to share them. Those would want to use them would benefit from early warnings.

Reply:
Ok, then maybe all we need is `test/scripts` and have a travis script ther if necessary.

Is it possible to follow the format of the scripts in the mfem/test repo? For example can we simply drop slightly modified versions of https://mybitbucket.llnl.gov/projects/MFEM/repos/tests/browse/documentation, https://mybitbucket.llnl.gov/projects/MFEM/repos/tests/browse/code-style and https://mybitbucket.llnl.gov/projects/MFEM/repos/tests/browse/gitignore in the new direcory? 

Note that https://mybitbucket.llnl.gov/projects/MFEM/repos/tests/browse/runtest provides a generic test driver that captures stdout and stderr which could also be useful...

Reply:
Not using the test driver yet. I kept things simple for this first PR. We should discuss further testing.